### DISCUSSION TITLE
Cloning failed after hellish night of misdirection and underdocumentation

### DISCUSSION THREAD

**Discussion by elmarsto** [vercel/community#3216] [Help]
### Summary

Hi all,

I'm attempting to deploy a NextJS project from a private GH repo, but I keep getting told that cloning failed (after a ten minute wait.)

I speculate that this is because I'd tried repeatedly to get this damn thing to deploy last night, racking up several dozen deploy attempts, but to no avail (the problem was that my `/public/image` directory was being included on Serverless Functions, which made them ~255MiB (well over the 50MiB limit imposed by AWS.) My best guess is that I've hit a GitHub soft limit (https://github.com/orgs/community/discussions/24841). **Could someone please confirm that this is the case?** If it is the case, it means that my best course of action is to wait. If it's not the case, I should be looking for a solution.  

The backstory is relevant as well, as **it points to a severe Developer Experience gotcha in Next.js deployment with Vercel,** presumably something that is antithetical to Vercel's market interest. 

Here's the backstory:

I was trying vainly for hours to get a vercel.json config that would exclude the `/public/` directories from my NextJS-based blog. Because it's a blog, these directories are full of high-res images (and some rather nice Butterick fonts,) and almost certainly the source of the bundle oversize, as the actual endpoint in question is a minimal graphql affair.  

I was up to my elbows in this effort only to discover a comment, deep downthread on a github issue, (https://github.com/orgs/vercel/discussions/103#discussioncomment-6262223) that the only working solution was actually `outputFileTracingExcludes` in `next.config.js`, because `vercel.json`'s `excludeFiles` is siliently (!?WTF) ignored specifically on NextJS projects (or so say the docs, when I read them more carefully: https://vercel.com/docs/concepts/projects/project-configuration#functions)

**Take-home message**:

  - There needs to be *way* more documentation about the 50MiB Serverless limitation. Mention of this should be prioritized in tutorials and all reference materials. It's bad DX to have developers go dig through issues on GH to even understand why the error they're being shown is being shown. The documentation linked to by the error message (https://vercel.com/docs/concepts/limits/overview#serverless-function-size) is **actively misleading** for NextJS users, again, because the `excludesFiles` option is silently ignored for NextJS. (https://vercel.com/docs/concepts/projects/project-configuration#functions)
  
 - There are no comprehensive examples shown of using `excludesFiles` in a `vercel.json`, and again, I mostly found out how to do this by reading GH issues. The docs mention the `excludesFiles` option only in passing.  *There is only one parenthetical comment at the end of the paragraph casually mentioning that the option does not work for NextJS, and it does not mention what a NextJS user is supposed to do instead*.  This is just... bad?

- The worst, worst, worst thing, though, is that there's *no 1p documentation whatseover* (that I have been able to find) that explains that you can resolve the AWS Function limit with `outputFileTracingExcludes` in `next.config.js`. This is an expectable encounter for anyone with even a modest number of assets

- Oh, also, while I'm at it, the Git LFS switch should be on by default in Vercel repos. There is no reason to leave devs in the merciless hands of mysterious build failures. It's not reasonable to expect that someone would be like, "I daresay my repo may be LFS; if it is, mightn't this cause a build error on Vercel, unless I explicitly enabled LFS there as well?" 
 
I'm going to create a set of issues linking back to this saga. 

### Steps to Reproduce

see above

**Comment by amyegan** [comment]
I'm sorry you had such a rough experience and I appreciate you taking the time to write down all of the details. I'm reading through all of it now, and I'll share all of this feedback with the appropriate teams.

As for your question about GitHub soft blocking, I don't have insight into their systems but I can look into any problems on the Vercel side if you share the project id.

**Reply by elmarsto** [reply]
Thanks Amy! The project id is lizmars-net. An example of a build that isn't working is lizmars-bj9udrq9i-1attice.vercel.app

**Reply by amyegan** [reply]
I checked the logs for this build and found a little more detail buried there: 
```
This repository is over its data quota. Account responsible for LFS bandwidth should purchase more data packs to restore access.
```

Cloning failed because the repo is over the [monthly LFS quota](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage) provided by GitHub. So it seems you were right about a soft block.

In other situations where cloning fails, I might recommend [deploying from local build](https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt). But I'm not sure that will work in this situation. 🙁

**Reply by elmarsto** [reply]
Thanks! While investigating, I found another bug, on GH this time. It would appear that the GitHub dashboard is buggy: 
![image](https://github.com/vercel/community/assets/315987/1781641d-e048-4119-8870-613ea7226081)


I've most certainly used up more than 0GB

**Reply by elmarsto** [reply]
While this bug is obviously not Vercel's to fix, you all should probably be aware that your GH customers won't be aware of their LFS bandwidth usage.

**Reply by elmarsto** [reply]
Here's a screengrab with more surrounding page context (better social proof) 
![image](https://github.com/vercel/community/assets/315987/8e91af74-ad04-4fb9-9970-3e0c7ab585d2)
